### PR TITLE
SUI: Canvas should be zoomable

### PR DESCRIPTION
**Pull request by dlabrecq**
Added functionality to make blueprint canvas zoomable.

https://patternfly.atlassian.net/browse/CFUX-278
https://www.pivotaltracker.com/n/projects/1914499/stories/135548487

Zoom Level: 100%
maxzoom

Zoom Level: 75%
midzoom

Zoom Level: 50%
minzoom

### CODE REVIEW

**Comment by dtaylor113** [comment]
Hi @dlabrecq , nice job!
Just wondering if the + - icons display when viewing a published blueprint?   Published blueprints are read-only so the duplicate and delete buttons are hidden (which might affect the alignment of the + - icons/buttons).  -thanks

**Comment by dlabrecq** [comment]
Yes, the icons should be available for a published blueprint. I saw the read-only layout in the HTML and tested it with and without the duplicate and delete buttons. The + and - icons still align right as expected.

**Comment by dtaylor113** [comment]
Awesome @dlabrecq , trying to test this branch, but can't log into SUI :frowning:  I did:
```
yarn cache clean
rm -rf node_modules
yarn run build
yarn start
```
Everything builds, runs fine.  But can't login.  Did you use these build steps?

**Comment by dtaylor113** [comment]
Got it running, some webpack dev settings.

Noticed that if a zoom out to max., then drop an item on the canvas all way over on right, then zoom back in, the item is no longer shown (as expected), however there is no vertical scroll bar to be able to scroll/pan over to see the offscreen item.

**Comment by dtaylor113** [comment]
Another issue -which I think was somewhat pre-existing, is that when I resize the browser, I don't see horiz. or vert. scrollbars.   I thought I had this working in the past, however wasn't ideal because as one resized browser the vert. scroll bar got smaller and smaller  -but it was there :smile:

**Comment by dtaylor113** [comment]
Code looks great!  :+1:

**Comment by dtaylor113** [comment]
Hmm...I somewhat take back my first comment about scrollbars when zooming in & out.  Turns out I had dragged the browser tab to a new screen, which has a lower resolution, so the bottom (horiz) scrollbar wasn't onscreen.   If I turn my display to 1280x720, screen only shows a small portion of SUI and both horiz and vert. scrollbars are not shown/offscreen.  
Think we need to determine what resolutions we support.

**Comment by dlabrecq** [comment]
Regarding the max zoom-out... Yes, that is expected. The canvas is the same size, but the underlying SVG is a bit larger to accommodate zooming out.

When you start out at 100%, you may notice that you can scroll. Depending on the OS, the scroll bars may not appear until you actually begin scrolling (e.g., with the Mac mouse).

As you mentioned, if you place a node to the far right (when zoomed out by 50%), then zoom in, you'll need to scroll to see it at the 100% zoom level. 

When you're max zoomed out by 50%, the canvas and SVG are the same size, so there is no scrolling at this zoom level.

That said, I reviewed this with Chris and he was ok with it. However, we can tweak this later if necessary. An alternative is to let the canvas shrink when zooming out. However, that doesn't provide any more room to work with.

**Comment by dtaylor113** [comment]
LGTM :-)

File: client/app/components/blueprints/blueprint-editor/blueprint-editor.directive.js [hunk 1]
```diff
@@ -209,5 +209,32 @@
...
+    //
+    vm.maxZoom = function() {
+      if (vm.chartViewModel && vm.chartViewModel.zoom) {
+        return vm.chartViewModel.zoom.isMax();
+      } else {
+        return false;
+      }
+    };
+
+    vm.minZoom = function() {
```
Comment (AllenBW) [inline client/app/components/blueprints/blueprint-editor/blueprint-editor.directive.js]: comment at line 217 also applies

File: client/app/components/blueprints/blueprint-editor/blueprint-editor.directive.js [hunk 2]
```diff
@@ -209,5 +209,32 @@
         })[0];
       }
     };
+
+    //
+    // Zoom
+    //
+    vm.maxZoom = function() {
+      if (vm.chartViewModel && vm.chartViewModel.zoom) {
```
Comment (AllenBW) [inline client/app/components/blueprints/blueprint-editor/blueprint-editor.directive.js]: Recommend simplifying to: 
```
vm.maxZoom = vm.chartViewModel && vm.chartViewModel.zoom ? vm.chartViewModel.zoom.isMax() : false;
```

**Comment by dlabrecq** [comment]
Simplified vm.maxZoom and vm.minZoom, but kept them as functions because the buttons are updated dynamically.

**Comment by chriskacerguis** [comment]
I checked Code Climate, and it passed, however the status in Github is still "pending" as of this comment.  Going to merge.

**Pull request merged**